### DISCUSSION TITLE
Originals: Thumbnails for RAW and video files not displaying - intended behaviour?

### DISCUSSION THREAD

**Discussion by dinoschachten** [photoprism/photoprism#2218] [Q&A] [answered]
Hi everyone,
I noticed that in the "Originals" view under "Library" you do get to look at the contents of folders in their original hierarchy, but you don't get thumbnails for non-JPG files such as videos or RAW files. I am wondering if this is intentional?
It seems like a bug (or an oversight) to me given that once I click on a RAW file with no thumbnail, I do get the rendered JPG version displayed - it's really only missing the thumbnails in the gallery.

I think this greatly diminishes the usability of an otherwise very helpful option of browsing (for instance in order to add all pictures from an existing folder to an album). Thumbnails in the "originals" section would be very helpful in order to go through your existing structure folder by folder, Note that the originals section has been cited as the answer to the request for a hierarchically displayed "folders" section, but without thumbnails, it does not serve that purpose.

This behaviour can also be observed in the official demo: https://demo.photoprism.app/library/files/demo

My environment:
- Ubuntu 20.04
- PhotoPrism installed via Docker (Build 220302-0059f429-Linux-AMD64)
- Thumbnails on local SSD
- Originals on NAS, mounted via SMB

Steps to reproduce:
- Install PhotoPrism
- Index your existing folders
- Navigate to Library > Originals > [a folder that contains RAW files]

**Comment by graciousgrey** [comment, marked as answer]
The originals section differs from the other views, such as the folder view.

The folder view (as well as all other views) shows "photos," whereas the originals section shows "files". 

A "photo" can consist of multiple files e.g. 
- raw + jpg + xmp 
- jpg + jpg 
- jpg  + json

The context menu actions also differ between files and photos. 

In the folder view in each case you see one photo (the primary file of the stack). 

In the originals section you see all files, only jpgs and videos have a thumbnail preview.  

We could add thumbnails to raw files as well (we might already have an issue for it) those will be the same as the ones for the related jpg files.

**Reply by dinoschachten** [reply]
Thanks for the explanation! I figured that would be the logic. Makes sense.

Indeed I think thumbnails would be very helpful to add more flexibility in regards to how users can utilise the _originals_ section.

However, I just realised I remembered the content of [issue 1536](https://github.com/photoprism/photoprism/issues/1536) incorrectly, with your final "verdict" being that the _folders_ section should provide a hierarchical view option (toggle) in the future.

For me personally both would do the trick - so I'd say: Whichever is easier to implement. ;)
